src/components/language-switch/language-switch.tsx

```diff
@@ -1,44 +1,115 @@
-  margin: 1px 0 1px 12px;
+interface LanguageSwitchProps {
+  translation: any;
+  className?: string;
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}
+
+const getTextColor = ({
+  fromNavigation,
+  $lightTheme,
+  theme,
+}: {
+  fromNavigation?: boolean;
+  $lightTheme?: boolean;
+  theme: DefaultTheme;
+}) => {
+  if (fromNavigation || $lightTheme) {
+    return theme.palette.text.default;
+  } else {
+    return theme.palette.text.header;
+  }
+};
+
+const getTextHoverColor = ({
+  fromNavigation,
+  $lightTheme,
+  theme,
+}: {
+  fromNavigation?: boolean;
+  $lightTheme?: boolean;
+  theme: DefaultTheme;
+}) => {
+  if (fromNavigation) {
+    return theme.palette.text.header;
+  } else if (!$lightTheme) {
+    return theme.palette.text.headerHover;
+  }
+};
+
+const StyledNav = styled.nav<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
+  transition: color 0.2s;
+  color: ${(props) => getTextColor(props)};
+  &:hover {
+    color: ${(props) => getTextHoverColor(props)};
+  }
+`;
+
+const StyledSelection = styled.select<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
+  background: none;
+  border: none;
+  text-transform: uppercase;
   font-weight: bold;
   font-size: 14px;
-  line-height: 110%;
+  color: inherit;
+
+  appearance: none;
+  padding-left: 10px;
   cursor: pointer;
-  text-transform: uppercase;
+`;
 
-  ${({ selected }) => (selected ? `color: #668CFF;` : `color: #FFFFFF;`)}
+export const StyledChevron = styled(Chevron)<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
+  margin-bottom: 2px;
+  /*margin-right: -6px makes the Chevron clickable*/
+  margin-right: -6px;
 `;
 
 export const LanguageSwitch = ({
-  translation,
   className = 'language-switch',
-}) => {
-  const { languages, language, originalPath, t } = useI18next();
+  $lightTheme,
+  fromNavigation,
+}: LanguageSwitchProps) => {
+  const { languages, language, t, changeLanguage } = useI18next();
 
   return (
-    <nav aria-label={t('navigation.language-aria')} className={className}>
-      {languages.map((languageOfLink) => {
-        return (
-          <StyledLanguageLink
-            key={languageOfLink}
-            to={
-              language !== languageOfLink && translation
-                ? translation
-                : originalPath
-            }
-            language={languageOfLink}
-            selected={language === languageOfLink}
-            title={t(`navigation.${languageOfLink}`)}
-          >
+    <StyledNav
+      aria-label={t('navigation.language-aria')}
+      className={className}
+      $lightTheme={$lightTheme}
+      fromNavigation={fromNavigation}
+    >
+      <StyledChevron
+        $lightTheme={$lightTheme}
+        fromNavigation={fromNavigation}
+      />
+      <StyledSelection
+        onChange={(event) => {
+          changeLanguage(event.target.value);
+        }}
+        value={language}
+        $lightTheme={$lightTheme}
+        fromNavigation={fromNavigation}
+      >
```

Comment:
If you agree to remove the generic types the props would also need to be removed. Less code means less bugs so I still suggest to do so.
```suggestion
      <StyledChevron />
      <StyledSelection
        onChange={(event) => {
          changeLanguage(event.target.value);
        }}
        value={language}
      >
```